userd/launcher_test.go

```diff
@@ -83,3 +93,98 @@ func (s *launcherSuite) TestOpenURLWithFailingXdgOpen(c *C) {
 	c.Assert(err, NotNil)
 	c.Assert(err, ErrorMatches, "cannot open supplied URL")
 }
+
+func (s *launcherSuite) TestOpenFileUserAccepts(c *C) {
+	mockZenity := testutil.MockCommand(c, "zenity", "true")
+	defer mockZenity.Restore()
+
+	path := filepath.Join(c.MkDir(), "test.txt")
+	c.Assert(ioutil.WriteFile(path, []byte("Hello world"), 0644), IsNil)
+
+	file, err := os.Open(path)
+	c.Assert(err, IsNil)
+	defer file.Close()
+
+	dupFd, err := syscall.Dup(int(file.Fd()))
```

Comment:
don't forget to close these fds

Reply:
`OpenFile` closes the file descriptor passed to it.  It seems godbus simply transfers ownership of the file descriptor to the method implementation, so I'm providing OpenFile with its own fd in the tests.